timely_beliefs/beliefs/classes.py

```diff
@@ -1416,6 +1434,13 @@ def resample_events(
             return self
         df = self
 
+        # Resample instantaneous sensors
```

Comment:
I can't explain why your new function only applies to instantaneous sensors. Can you add that?

And why is that the one case in which we take so much care of DST transitions?  (this `resample_events`  function here has two other cases)

Note: The NB about `keep_only_most_recent_belief=True` applies to only one case, the reader might assume it's about all of them.